nadlogar/problems/models/linearna.py

```diff
@@ -0,0 +1,690 @@
+        y = sympy.symbols('y')
+        (koeficient_x, koeficient_y, prosti, implicitnaOblika) = implicitna_premica()
+
+        implicitna = sympy.latex(sympy.Eq(implicitnaOblika, 0))
+        eksplicitna = sympy.latex(sympy.Eq(y,
+                                           sympy.Rational(-koeficient_x, koeficient_y) * x + sympy.Rational(-prosti, koeficient_y), evaluate=False))
+        odsekovna = sympy.latex(
+            sympy.Eq(sympy.Rational(-koeficient_x, prosti) * x + sympy.Rational(-koeficient_y, prosti) * y, 1))
+
+        return {'implicitna': implicitna, 'eksplicitna': eksplicitna, 'odsekovna': odsekovna}
+
+
+class PremiceTrikotnik(Problem):  # TODO preveri jinja latex
+    """
+    Naloga za računanje ploščine trikotnika, ki ga dve premici oklepata z abscisno osjo.
+    .. runblock:: pycon
+        >>> import sys
+        >>> sys.path.append('../')
+        >>> from linearna_funkcija import *
+        >>> PremiceTrikotnik().primer()
+    .. runblock:: pycon
+        >>> import sys
+        >>> sys.path.append('../')
+        >>> from linearna_funkcija import *
+        >>> PremiceTrikotnik().primer()
+    """
+    default_instruction = r"Izračunaj ploščino trikotnika, ki ga premici $@premica1$ in $@premica2$ oklepata z abscisno osjo."
+    besedilo_vecih = r''' Izračunaj ploščino trikotnika, ki ga premici oklepata z abscisno osjo:
+    \begin{enumerate}
+    {% for naloga in naloge %}
+    \item ${{@premica1}}$ in ${{@premica2}}$
+    {% endfor %}
+    \end{enumerate}
+    '''
+    default_solution = r"$S=@ploscina$"
+    resitev_vecih = r'''
+    \begin{enumerate}
+     {% for naloga in naloge %}
+     \item $S={{@ploscina}}$
+     {% endfor %}
+     \end{enumerate}
+     '''
+
+    class Meta:
+        verbose_name = "Linearna funkcija / Ploščina trikotnika"
+
+    def generate(self):
+        x1 = izberi_koordinato(1, 5)
+        y1 = 0
+        x2 = random.choice([x for x in range(-5, 6) if x != 0])
+        y2 = random.choice([x for x in range(-5, 6) if x != 0])
+        x3 = izberi_koordinato(-5, 0)
+        y3 = 0
+        x = sympy.symbols('x')
+        y = sympy.symbols('y')
+        if not (x2 != x1 and x2 != x3):  # premici sta vzporedni
+            raise GeneratedDataIncorrect
+        if x2 == x1:
+            premica1 = sympy.Eq(x, x1)
+        else:
+            k1 = sympy.Rational((y2 - y1), (x2 - x1))
+            n1 = y1 - k1 * x1
+            premica1 = sympy.Eq(y, k1 * x + n1)
+        if x3 == x2:
+            premica2 = sympy.Eq(x, x3)
+        else:
+            k2 = sympy.Rational((y3 - y2), (x3 - x2))
+            n2 = y3 - k2 * x3
+            premica2 = sympy.Eq(y, k2 * x + n2)
+
+        ploscina_trikotnika = sympy.latex(
+            sympy.simplify(abs((x2 - x1) * (y3 - y1) - (x3 - x1) * (y2 - y1)) / 2).evalf(3)).rstrip('0').rstrip('.')
+
+        return {'premica1': sympy.latex(premica1), 'premica2': sympy.latex(premica2), 'ploscina': ploscina_trikotnika}
+
+
+class NarisiLinearnoFunkcijo(Problem):
```

Comment:
Naloge z risanjem grafov povsod zakomentiraj, ker jih zanekrat še ne moremo risati.

Reply:
Naloge bi verjetno lahko tudi pustili in zaenkrat pač še ne bi vračale rešitev (zakomentiramo samo rešitve, dokler ne ugotovimo, kako bomo risali grafe).

Reply:
Problem je v tem, da če se ne motim, to ne bi delovalo. Mislim da bi se sprožila napaka v testih. Poleg tega grdo izgleda na spletni strani.
Seveda pa bomo "oživili" te naloge, potem ko nam uspe ustvariti risanje grafov.